clarity_ext/utility/testing_parse_scripts/builders.py

```diff
@@ -0,0 +1,122 @@
+from mock import MagicMock
+from StringIO import StringIO
+from io import BytesIO
+from clarity_ext.service.file_service import Csv
+from clarity_ext.service.file_service import FileService
+from clarity_ext.service.process_service import ProcessService
+from clarity_ext.context import ExtensionContext
+from clarity_ext.service.artifact_service import ArtifactService
+from clarity_ext.domain.process import Process
+from clarity_ext.domain.user import User
+from clarity_ext.domain.udf import UdfMapping
+from clarity_ext.utility.testing_parse_scripts.fake_artifact_factory import FakeArtifactFactory
+
+
+class ContextBuilder:
+    """
+    Add entities to repositories held by context
+    E.g. shared files, artifacts, analytes.
+    """
+    def __init__(self):
+        self.step_repo = FakeStepRepo()
+        self.logger = FakeLogger()
+        self.context = None
+        self._create()
+
+    def with_analyte_pair(self, input, output):
+        self.step_repo.add_analyte_pair(input, output)
+
+    def with_mocked_local_shared_file(self, filename, contents):
+        monkey = LocalSharedFilePatcher()
+        monkey.cache[filename] = contents
+        self.context.local_shared_file = monkey.local_shared_file
+
+    def _create(self):
+        session = MagicMock()
```

Comment:
When creating the mock objects I think it might be worth using `create_autospec` (https://docs.python.org/3/library/unittest.mock.html#unittest.mock.create_autospec), which will decrease the likelyhood that the tests fail for the wrong reason. I.e. when just using a `MagicMock`, every method called will return an instance of `MagicMock`, even if that method is not defined on the object it tries to mock. If autospecing is used, it will thrown an exception if a method that does not exist on the specified object is called.